src/output/jld2_output.jl

```diff
@@ -0,0 +1,153 @@
+export JLD2Output
+
+"""Output writer for a JLD2 file that saves the PrognosticVariables
+and DiagnosticVariables structs directly to a JLD2 file. All internal
+scalings and units are still applied to these outputs. Fields are 
+$(TYPEDFIELDS)"""
+@kwdef mutable struct JLD2Output <: AbstractOutput
+
+    # FILE OPTIONS
+    active::Bool = false
+    
+    "[OPTION] path to output folder, run_???? will be created within"
+    path::String = pwd()
+    
+    "[OPTION] run identification number/string"
+    id::String = "0001"
+    run_path::String = ""                   # will be determined in initalize!
+    
+    "[OPTION] name of the output jld2 file"
+    filename::String = "output.jld2"
+    
+    "[OPTION] also write restart file if output==true?"
+    write_restart::Bool = true
+    pkg_version::VersionNumber = isnothing(pkgversion(SpeedyWeather)) ? v"0.0.0" : pkgversion(SpeedyWeather)
+
+    "[OPTION] output frequency, time step"
+    output_dt::Second = Second(DEFAULT_OUTPUT_DT)
+
+    "[OPTION] will reopen and resave the file to merge everything in one big vector. Turn off if the file is too large for memory."
+    merge_output::Bool = true
+
+    "[OPTION] output the PrognosticVariables" 
+    output_prognostic::Bool = true 
+
+    "[OPTION] output the DiagnosticVariables as well" 
+    output_diagnostic::Bool = true 
+
+    # TIME STEPS AND COUNTERS (initialize later)
+    output_every_n_steps::Int = 0           # output frequency
+    timestep_counter::Int = 0               # time step counter
+    output_counter::Int = 0                 # output step counter
+
+    jld2_file::Union{JLDFile, Nothing} = nothing
```

Comment:
We have the same field in `NetCDFOutput` but only really exists because I didn't know how to create an empty `NetCDFFile` as we can't really create this at the time when `NetCDFOutput` is created -- hence the mutable struct and put in there something so that we can overwrite it later, feels a bit hacky if you know better, feel free to change that!